### PR TITLE
Set the default of skipSelf to true

### PR DESCRIPTION
**Pull request by TrickyPi**
This PR contains:

- [ ] bugfix
- [ ] feature
- [x] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [x] yes (_breaking changes will not be merged unless absolutely necessary_)
- [ ] no

List any relevant issue numbers:

### Description
The relevant task https://github.com/orgs/rollup/projects/1/views/1?pane=issue&itemId=12703375.

### CODE REVIEW

File: docs/plugin-development/index.md [hunk 1]
```diff
@@ -508,12 +508,8 @@ function injectPolyfillPlugin() {
...
 			}
 			if (options.isEntry) {
-				// Determine what the actual entry would have been. We need
-				// "skipSelf" to avoid an infinite loop.
-				const resolution = await this.resolve(source, importer, {
-					skipSelf: true,
-					...options
-				});
+				// Determine what the actual entry would have been.
+				const resolution = await this.resolve(source, importer, options);
```
Comment (lukastaegert) [inline docs/plugin-development/index.md:512]: Good thinking to also update documentation!

File: docs/plugin-development/index.md [hunk 2]
```diff
@@ -1699,7 +1692,7 @@ The return type **ResolvedId** of this hook is defined in [`this.getModuleInfo`]
...
+The default of `skipSelf` is `true`, So the `resolveId` hook of the plugin from which `this.resolve` is called will be skipped when resolving. When other plugins themselves also call `this.resolve` in their `resolveId` hooks with the _exact same `source` and `importer`_ while handling the original `this.resolve` call, then the `resolveId` hook of the original plugin will be skipped for those calls as well. The rationale here is that the plugin already stated that it "does not know" how to resolve this particular combination of `source` and `importer` at this point in time. If you do not want this behaviour, do not use `skipSelf` but implement your own infinite loop prevention mechanism if necessary.
```
Comment (lukastaegert) [inline docs/plugin-development/index.md:1695]:
```suggestion
The default of `skipSelf` is `true`, So the `resolveId` hook of the plugin from which `this.resolve` is called will be skipped when resolving. When other plugins themselves also call `this.resolve` in their `resolveId` hooks with the _exact same `source` and `importer`_ while handling the original `this.resolve` call, then the `resolveId` hook of the original plugin will be skipped for those calls as well. The rationale here is that the plugin already stated that it "does not know" how to resolve this particular combination of `source` and `importer` at this point in time. If you do not want this behaviour, set `skipSelf` to `false` and implement your own infinite loop prevention mechanism if necessary.
```

**Review by lukastaegert — APPROVED** [review]
Just a minor wording change, looks good to me!

File: docs/plugin-development/index.md (hunk 2 above)
Comment (TrickyPi) [reply docs/plugin-development/index.md]: Thanks for your thorough review!

**Pull request merged**